### DISCUSSION TITLE
Can the module store new files via a command?

### DISCUSSION THREAD

**Discussion by adamjmoritz** [Makuna/DFMiniMp3#108] [Q&A]
I'm wondering if I can use wi-fi to write/receive a new mp3 file to the SD card on the module.  

I'd like the ability to update files dynamically for my project.

Or is the SD card/slot on the "MP3-TF-16P/MH2024K-16SS" only a static read/only memory device?

**Comment by Makuna** [comment]
Nope.  The exposed API has nothing about being able to read or write files.

**Reply by adamjmoritz** [reply]
Thanks for the response.  Yes, I did find the API and noticed that.  I just wonder if there is a way to do it via external (via Arduino uno) pin controls.  I probably should have phrased my question better.   Or, perhaps the ability to play from an online source rather than the SD card?

**Reply by Makuna** [reply]
This library exposes every documented feature through the program interface. It's a very limited functionality device.

There is the `DfMp3_PlaySource` enumeration that is used as an argument which defines the source of playback (USB, SD, AUX, FLASH).  The `getPlaySources()` does not even work on your specific chip though.  I have no clue as to how AUX would work or connect; and I don't know if there was ever made a version with FLASH.  I have never tried to connect the USB so I don't know what external hardware is needed.

I guess you could try to hack it to physically share the SD card and have the Arduino write directly to it.  But that's beyond the topic of this library.

**Comment by adamjmoritz** [comment]
Ok, great information.  I'll experiment with the USB, hopefully it works.
